gnuradio-runtime/include/gnuradio/basic_block.h

```diff
@@ -28,6 +28,28 @@
 
 namespace gr {
 
+class msg_queue_comparator
+{
+private:
+    const pmt::pmt_t d_system_port = pmt::intern("system");
+
+public:
+    bool operator()(pmt::pmt_t const& queue_key1, pmt::pmt_t const& queue_key2) const
+    {
+        bool is_system1 = pmt::eqv(queue_key1, d_system_port);
+        bool is_system2 = pmt::eqv(queue_key2, d_system_port);
+
+        if (is_system1 && is_system2)
+            return false;
+        else if (is_system1 && !is_system2)
+            return true;
+        else if (!is_system1 && is_system2)
+            return false;
+        else
+            return queue_key1 < queue_key2;
```

Comment:
How is this comparison defined? Would it be an option to use `pmt::comparator`?

Reply:
I incorrectly assumed it would use the comparator. I'll change the code to explicitly call it.